### PR TITLE
feat: improve routes, refactoring, adapt pagestructure, adapt tests (DEV-1670)

### PR DESCRIPTION
**Pull request by Vijeinath**
resolves DEV-1670

### CODE REVIEW

File: src/app/services/cpe-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * The interface for getting projects from the api service
+ */
+export interface iPage extends iCpeResource {
+    gridDimension: Dimension
+    header: Widget[];
+    body: Widget[];
+    footer: Widget[]
+}
+
+type Widget = {
```
Comment (domsteinbach) [inline src/app/services/cpe-api.service.ts:37]: Why is it a type here? I would prefer to have it streamlined with iProject asf. as an interface like so:

export interface iPage extends iCpeResource {
    gridDimension: Dimension
    header: Widget[];
    body: Widget[];
    footer: Widget[];
}

**Review by domsteinbach — CHANGES_REQUESTED** [review]
LGTM. NgOnChanges, type annotations, bit lint and maybe the interface instead of the type (see comments).

File: src/app/services/project-repository.service.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            .pipe(
+                map((project: iProject) =>
+                    new Project(project.id, project.label, project.description, project.mainPageID))
+            );
+    }
+
+    /**
+     * get a project via the api service. Return a Project as Promise.
+     */
+    getProjectByIdFull(id: string): Observable<any> {
```
[the remark this answers is not in the document]
Comment (Vijeinath) [reply src/app/services/project-repository.service.ts]: Done in https://github.com/dasch-swiss/dsp-cpe/pull/15/commits/542b92bf90b7c9b3e8f6c3f7cb585a0ba1c912c7

File: src/app/playground/playground.module.ts [hunk 3]
```diff
@@ -1,20 +1,19 @@
-import { NgModule } from '@angular/core';
-import { CommonModule } from '@angular/common';
-import { CpePlaygroundComponent } from './cpe-playground/cpe-playground.component';
-import { GuiElementsModule } from "../gui-elements-module/gui-elements.module";
-
-
+import {NgModule} from "@angular/core";
```
[the remark this answers is not in the document]
Comment (Vijeinath) [reply src/app/playground/playground.module.ts:1]: I agree. I would suggest we create an issue for this and tackle it later

File: src/app/routing-module/app-routing.module.ts [hunk 4]
```diff
@@ -1,39 +1,35 @@
-import { NgModule } from '@angular/core';
-import { RouterModule, Routes } from '@angular/router';
+import {NgModule} from "@angular/core";
```
[the remark this answers is not in the document]
Comment (Vijeinath) [reply src/app/routing-module/app-routing.module.ts:1]: I agree. I would suggest we create an issue for this and tackle it later

File: src/app/cpe/project/project.component.ts [hunk 5]
```diff
@@ -1,67 +1,80 @@
...
 @Component({
-  selector: 'app-project',
-  templateUrl: './project.component.html'
+    selector: "app-project",
+    templateUrl: "./project.component.html"
 })
 export class ProjectComponent implements OnInit, OnDestroy {
 
-  project: Project | undefined = undefined;
+    project: any | undefined = undefined;
```
[the remark this answers is not in the document]
Comment (Vijeinath) [reply src/app/cpe/project/project.component.ts]: Done https://github.com/dasch-swiss/dsp-cpe/pull/15/commits/7c575e57fc3b40a01c1b64bada30e04a0e42264b

File: src/app/services/cpe-api.service.ts (hunk 1 above)
Comment (Vijeinath) [reply src/app/services/cpe-api.service.ts]: Good thougth: https://github.com/dasch-swiss/dsp-cpe/pull/15/commits/00a40d88d5d09301501a7806d30725e5ea76d96b

**Review by domsteinbach — APPROVED** [review]
Kewl, LGTM

**Pull request merged**